src/presetsController.ts

```diff
@@ -1943,26 +1943,64 @@ export class PresetsController {
         return vscode.window.showTextDocument(vscode.Uri.file(presetsFilePath));
     }
 
+    // this is used for the file watcher on adding a new presets file
+    async onCreatePresetsFile() {
+        await this.reapplyPresets();
+        await this.project.projectController?.updateActiveProject(this.workspaceFolder);
+    }
+
     private async watchPresetsChange() {
-        if (this._presetsWatcher) {
-            this._presetsWatcher.close().then(() => {}, () => {});
-        }
 
-        const handler = () => {
+        const presetChangeHandler = () => {
             void this.reapplyPresets();
         };
-        this._presetsWatcher = chokidar.watch(this._referencedFiles, { ignoreInitial: true })
-            .on('add', handler)
-            .on('change', handler)
-            .on('unlink', handler);
+        const presetCreatedHandler = () => {
+            void this.onCreatePresetsFile();
+        };
+
+        const events: Map<string, () => void> = new Map<string, () => void>([
+            ["change", presetChangeHandler],
+            ["unlink", presetChangeHandler],
+            ["add", presetCreatedHandler]
+        ]);
+
+        this._presetsWatchers?.dispose();
+        this._presetsWatchers = new FileWatcher(this._referencedFiles, events, { ignoreInitial: true });
     };
 
     dispose() {
-        if (this._presetsWatcher) {
-            this._presetsWatcher.close().then(() => {}, () => {});
-        }
+        this._presetsWatchers?.dispose();
+
         if (this._sourceDirChangedSub) {
             this._sourceDirChangedSub.dispose();
         }
     }
 }
+
+/**
+ * FileWatcher is a wrapper around chokidar's FSWatcher that allows for watching multiple paths.
+ * Chokidar's support for watching multiple paths is currently broken, if it is fixed in the future, this class can be removed.
+ */
+class FileWatcher {
```

Comment:
If we are implementing `dispose`, we should likely about extending vscode.Disposable.

Reply:
I see that PresetsController also has dispose() but it doesnt extend vscode.Disposable - is there a difference here?

Reply:
FIrst, my mistake, I meant `implements`. And I would actually think that we should be doing it there too, because it's what stores the watcher

Reply:
https://vscode-api.js.org/classes/vscode.Disposable.html

The theory is that our wrapper contains a watcher/event listener (disposable), and then the controller holds that wrapper

Reply:
It's not a major difference in implementation, and it won't make a huge difference in performance, but it's technically correct and if we ever get high up enough that we can add it to the vscode list of disposables, then vscode could handle disposal when it terminates. So I would just add `implements vscode.Disposable` and leave it at that.